x-pack/plugins/observability_solution/dataset_quality/public/state_machines/dataset_quality_controller/src/state_machine.ts

```diff
@@ -780,22 +821,89 @@ export const createDatasetQualityControllerStateMachine = ({
         assertBreakdownFieldEcsFailedNotifier(toasts, event.data),
     },
     services: {
-      loadDataStreamStats: (context) =>
-        dataStreamStatsClient.getDataStreamsStats({
-          type: context.type as GetDataStreamsStatsQuery['type'],
-          datasetQuery: context.filters.query,
-        }),
-      loadDegradedDocs: (context) => {
-        const { startDate: start, endDate: end } = getDateISORange(context.filters.timeRange);
+      loadDataStreamStats:
+        (context, _event, { data: { type } }) =>
+        async (send) => {
+          try {
+            const dataStreamStats = await (isTypeSelected(type, context)
+              ? dataStreamStatsClient.getDataStreamsStats({
+                  type,
+                  datasetQuery: context.filters.query,
+                })
+              : Promise.resolve({
+                  dataStreamsStats: [],
+                  datasetUserPrivileges: {
+                    canMonitor: true,
+                    canRead: true,
+                    canViewIntegrations: true,
+                  },
+                }));
 
-        return dataStreamStatsClient.getDataStreamsDegradedStats({
-          type: context.type as GetDataStreamsStatsQuery['type'],
-          datasetQuery: context.filters.query,
-          start,
-          end,
-        });
-      },
+            send({
```

Comment:
why are we firing events from here and not using the `onDone` and `onError` handler in the state machine. We can definitely do this, but we are already using the onDone and onError style everywhere else

Reply:
The problem with `onDone` and `onError` comes when you invoke the same service multiple times. All the previous requests get cancelled so the event are triggered only for the last call you perform

Reply:
Sorry but i didn't understand the problem with that. Isn't that the expectation also in our case that we want only the last call value ?
1. Are you calling the same service multiple times ?

Reply:
ok 👍🏼